src/js/lib.js

```diff
@@ -726,16 +709,15 @@ vjs.parseUrl = function(url) {
  * @private
  */
 function _logType(type, args){
-  var argsArray, noop, console;
-
   // convert args to an array to get array functions
-  argsArray = Array.prototype.slice.call(args);
+  let argsArray = Array.prototype.slice.call(args);
   // if there's no console then don't try to output messages
   // they will still be stored in vjs.log.history
   // Was setting these once outside of this function, but containing them
   // in the function makes it easier to test cases where console doesn't exist
-  noop = function(){};
-  console = window['console'] || {
+  let noop = function(){};
```

Comment:
Is there a benefit to that style of noop? It seems borderline too clever to me, since it's kinda obtuse at a glance.

As an aside that shouldn't be taken into consideration *at all*, I did a quick search to see if there were other benefits I wasn't aware of and came across a [relevant JSPerf](http://jsperf.com/noop-vs-function-prototype). Clearly take with a few thousand massive grains of salt, but the orange is what we have (labeled "noop"), and red is what you're proposing (labeled "function prototype"). In everything except FF 34 where it blows everything out of the water by a mile (wat), the function prototype method is the slowest.